python/dgl/data/tu.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import absolute_import
+import numpy as np
+import dgl
+import os
+
+from dgl.data.utils import download, extract_archive, get_download_dir
+
+
+class TUDataset():
+    """
+    TUDataset contains lots of graph kernel datasets for graph classification.
+    Use provided node feature by default. If no feature provided, use one-hot node label instead.
+    If neither labels provided, use constant for node feature.
+
+    :param name: Dataset Name, such as `ENZYMES`, `DD`, `COLLAB`
+    :param use_pandas: Default: False.
+        Numpy's file read function has performance issue when file is large,
+        using pandas can be faster.
+    :param hidden_size: Default 10. Some dataset doesn't contain features.
+        Use constant node features initialization instead, with hidden size as `hidden_size`.
+
+    """
+
+    _url = r"https://ls11-www.cs.uni-dortmund.de/people/morris/graphkerneldatasets/{}.zip"
+
+    def __init__(self, name, use_pandas=False, hidden_size=10):
+
+        self.name = name
+        self.hidden_size = hidden_size
+        self.extract_dir = self._download()
+
+        if use_pandas:
+            import pandas as pd
+            DS_edge_list = self._idx_from_zero(
+                pd.read_csv(self._file_path("A"), delimiter=",", dtype=int).values)
+        else:
+            DS_edge_list = self._idx_from_zero(
+                np.genfromtxt(self._file_path("A"), delimiter=",", dtype=int))
+
+        DS_indicator = self._idx_from_zero(
+            np.genfromtxt(self._file_path("graph_indicator"), dtype=int))
+        DS_graph_labels = self._idx_from_zero(
+            np.genfromtxt(self._file_path("graph_labels"), dtype=int))
+
+        g = dgl.DGLGraph()
+        g.add_nodes(DS_edge_list.max() + 1)
+        g.add_edges(DS_edge_list[:, 0], DS_edge_list[:, 1])
+        g.add_edges(DS_edge_list[:, 1], DS_edge_list[:, 0])
+
+        node_idx_list = []
+        for idx in range(np.max(DS_indicator) + 1):
+            node_idx = np.where(DS_indicator == idx)
+            node_idx_list.append(node_idx[0])
+        self.graph_lists = g.subgraphs(node_idx_list)
+        self.graph_labels = DS_graph_labels
+
+        try:
+            DS_node_labels = self._to_onehot(
+                self._idx_from_zero(np.loadtxt(self._file_path("node_labels"), dtype=int)))
```

Comment:
I feel it's better not convert to onehot because many loss functions directly use integer labels.

Reply:
Added integer node label to `node_label` field for potential uses. And still use one-hot label as feature if node attribute is not available.